### PR TITLE
Apply new texture APIs

### PR DESCRIPTION
**Pull request by bbrto21**
* Resolves flutter-tizen/engine/issues/285.
* This change depends on flutter-tizen/engine/pull/338.

Signed-off-by: Boram Bae <boram21.bae@samsung.com>

### CODE REVIEW

File: packages/camera/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 0.3.3
+* Appaly new texture APIs.
```
[the remark this answers is not in the document]
Comment (bbrto21) [reply packages/camera/CHANGELOG.md]: Thanks!

File: packages/camera/tizen/src/camera_device.h [hunk 2]
```diff
@@ -329,7 +329,8 @@ class CameraDevice {
   long texture_id_{0};
   flutter::PluginRegistrar *registrar_{nullptr};
   std::unique_ptr<flutter::TextureVariant> texture_variant_;
-  std::unique_ptr<FlutterDesktopGpuBuffer> flutter_desktop_gpu_buffer_;
+  std::unique_ptr<FlutterDesktopGpuSurfaceDescriptor>
+      flutter_desktop_gpu_buffer_;
```
[the remark this answers is not in the document]
Comment (bbrto21) [reply packages/camera/tizen/src/camera_device.h]: I chose the latter and applied it.

File: packages/camera/tizen/src/camera_device.cc [hunk 3]
```diff
@@ -345,9 +346,11 @@ CameraDevice::CameraDevice(flutter::PluginRegistrar *registrar,
               current_packet_ = nullptr;
               return nullptr;
             }
-            flutter_desktop_gpu_buffer_->buffer = surface;
+            flutter_desktop_gpu_buffer_->handle = surface;
             flutter_desktop_gpu_buffer_->width = width;
+            flutter_desktop_gpu_buffer_->visible_width = width;
             flutter_desktop_gpu_buffer_->height = height;
+            flutter_desktop_gpu_buffer_->visible_height = height;
```
[the remark this answers is not in the document]
Comment (bbrto21) [reply packages/camera/tizen/src/camera_device.cc]: I removed them from all over this PR.

File: packages/camera/tizen/src/camera_device.cc [hunk 4]
```diff
@@ -345,20 +346,19 @@ CameraDevice::CameraDevice(flutter::PluginRegistrar *registrar,
...
+              CameraDevice *cd = (CameraDevice *)release_context;
+              cd->ReleaseMediaPacket();
+            };
+            gpu_surface_->release_context = this;
+            return gpu_surface_.get();
           }));
   texture_id_ =
       registrar_->texture_registrar()->RegisterTexture(texture_variant_.get());
-  flutter_desktop_gpu_buffer_ = std::make_unique<FlutterDesktopGpuBuffer>();
+  gpu_surface_ = std::make_unique<FlutterDesktopGpuSurfaceDescriptor>();
```
Comment (swift-kim) [inline packages/camera/tizen/src/camera_device.cc:361]: `FlutterDesktopGpuSurfaceDescriptor` is a simple struct type. Should this value managed as a unique pointer?

File: packages/camera/tizen/src/camera_device.cc [hunk 5]
```diff
@@ -345,20 +346,19 @@ CameraDevice::CameraDevice(flutter::PluginRegistrar *registrar,
...
-                  CameraDevice *cd = (CameraDevice *)release_context;
-                  cd->ReleaseMediaPacket();
-                };
-            flutter_desktop_gpu_buffer_->release_context = this;
-            return flutter_desktop_gpu_buffer_.get();
+            gpu_surface_->handle = surface;
+            gpu_surface_->width = width;
+            gpu_surface_->height = height;
+            gpu_surface_->release_callback = [](void *release_context) {
+              CameraDevice *cd = (CameraDevice *)release_context;
```
Comment (swift-kim) [inline packages/camera/tizen/src/camera_device.cc:353]: We may need another PR to do some refactoring (to fix the C type cast).

File: packages/camera/tizen/src/camera_device.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (swift-kim) [reply packages/camera/tizen/src/camera_device.cc]: Okay, we still keep `width` and `height` although they are unused.

File: packages/camera/tizen/src/camera_device.cc (hunk 5 above)
Comment (bbrto21) [reply packages/camera/tizen/src/camera_device.cc:353]: I also think other PR is better.

File: packages/camera/tizen/src/camera_device.cc (hunk 4 above)
Comment (bbrto21) [reply packages/camera/tizen/src/camera_device.cc:361]: > I mean, the value doesn't need to be a member of the class.

Did you mean that we don't need to have a pointer of `FlutterDesktopGpuSurfaceDescriptor` as member?
I'd appreciate it if you could give me a little more detail.

Comment (swift-kim) [reply packages/camera/tizen/src/camera_device.cc:361]: Ah, never mind. I thought `GpuSurfaceTexture`'s `ObtainDescriptorCallback` returns `FlutterDesktopGpuSurfaceDescriptor` by value so we can initialize an instance of `FlutterDesktopGpuSurfaceDescriptor` inside the callback.

Still we can convert `gpu_surface_` into a plain struct (not unique pointer) but it shouldn't matter much.

**Review by swift-kim — APPROVED** [review]

File: packages/camera/tizen/src/camera_device.cc (hunk 4 above)
Comment (bbrto21) [reply packages/camera/tizen/src/camera_device.cc:361]: thanks! :)

**Comment by bbrto21** [comment]
`no merge` label will be removed when flutter-tizen updated with latest engine.

**Pull request merged**